cpp/perspective/src/cpp/sparse_tree.cpp

```diff
@@ -2020,6 +2014,28 @@ t_stree::get_aggregate(t_index idx, t_index aggnum) const {
     return extract_aggregate(m_aggspecs[aggnum], c, agg_ridx, agg_pridx);
 }
 
+t_tscalar
+t_stree::get_aggregate_median(std::vector<t_tscalar>& values) const {
+    int size = values.size();
+    bool is_even_size = size % 2 == 0;
+
+    if (is_even_size && values[0].is_numeric()){
+        t_tscalar median_average;
+        std::vector<t_tscalar>::iterator first_middle = values.begin() + ((size - 1) / 2);
+        std::vector<t_tscalar>::iterator second_middle = values.begin() + (size / 2);
+
+        nth_element(values.begin(),  first_middle, values.end());
```

Comment:
`nth_element()` does not need to be called twice here, the column is guaranteed to be even and min 2, so this is equivalent to `*(second_middle - 1)`.

Reply:
Done!